### PR TITLE
lnrpc+itest: return channel Memo for Pending channels

### PR DESCRIPTION
**Pull request by shaurya947**
## Change Description
In a previous PR #7668  we added a Memo field for channels that could be specified when opening a channel. This was a reference note-to-self with no bearing on the functioning of the channel. In that PR, the memo value was returned only through ListChannels. This PR builds upon that PR by also returning the Memo field for pending open and waiting close channels.

## Steps to Test
An itest has been updated to verify that the Memo is returned but to manually test, one can verify the Memo field on a pending-open or waiting-close channel

### Testing
- [ ] Your PR passes all CI checks.
- [X] Tests covering the positive and negative (error paths) are included.
- [ ] Bug fixes contain tests triggering the bug to prevent regressions.

### Code Style and Documentation
- [X] The change obeys the [Code Documentation and Commenting](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#CodeDocumentation) guidelines, and lines wrap at 80.
- [X] Commits follow the [Ideal Git Commit Structure](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md#IdealGitCommitStructure).
- [X] Any new logging statements use an appropriate subsystem and logging level.
- [X]  [There is a change description in the release notes](https://github.com/lightningnetwork/lnd/tree/master/docs/release-notes), or `[skip ci]` in the commit message for small changes.

📝 Please see our  [Contribution Guidelines](https://github.com/lightningnetwork/lnd/blob/master/docs/code_contribution_guidelines.md) for further guidance.

### CODE REVIEW

**Comment by shaurya947** [comment]
Hey @guggero and @positiveblue I created this follow-up PR to add the Memo field to the PendingChannels response. To avoid bloating the itest suite (as was recommended in the last PR), I've added small bit of extra logic to an existing itest. I also haven't edited release notes—I read over the note from the last PR and I think it sufficiently covers this change as well. Let me know if all that is ok or if I should do something differently.

One thing to note is that PendingChannels returns three types of channels:
1. pending open
2. pending force close
3. waiting close

(1) and (3) fetch from the `openChannelBucket` and populate using `channeldb.OpenChannel` instance so the Memo field is present and all we do is pass it along in the protobuf message. (2) on the other hand fetches from the `closedChannelBucket` and populates using `channeldb.ChannelCloseSummary` so the Memo field is not available (neither in the bucket nor in the deserialized struct).

Should we update the code that marks a channel as closed (thereby moving it from the open -> closed bucket) to also serialize the Memo in `closedChannelBucket`? And should we update the `ChannelCloseSummary` when deserializing for (2)? I guess the underlying question is at what point does the Memo stop being useful? If the answer is "never" then we should probably also return it in ClosedChannels?

**Comment by guggero** [comment]
> Let me know if all that is ok or if I should do something differently.

No entry in the release notes is fine, I added the `no-changelog` label that makes the CI ignore that step (will work for the next push).

>  If the answer is "never" then we should probably also return it in ClosedChannels?

I think the memo is definitely helpful for pending force closed or fully closed channels. So adding it to the `ChannelCloseSummary` is a good idea. I don't know if we already have TLV values there, so it might be a bit more involved.

**Review by yyforyongyu — CHANGES_REQUESTED** [review]
I think the `skip ci` in the commit message needs to be removed to let the build run.

**Comment by shaurya947** [comment]
> I think the `skip ci` in the commit message needs to be removed to let the build run.

Aah my bad I thought the flag was necessary if I didn't update release notes. Will remove it.

@guggero I realized that the full state of the channel does get written to the `historicalChannelBucket` which is actually available when querying the pending force closing channel.

https://github.com/lightningnetwork/lnd/blob/bbbf7d33fb1527acebb44e2a69d16fbcf24cc2fa/rpcserver.go#L3499

This allows me to populate the Memo in all kinds of channels returned by `PendingChannels`. I also verified in the same itest that the Memo is indeed returned even when the channel transitions to the pending force closing state :slightly_smiling_face:

File: itest/lnd_onchain_test.go [hunk 1]
```diff
@@ -466,7 +466,8 @@ func testAnchorThirdPartySpend(ht *lntest.HarnessTest) {
 	// fully.
 	aliceChanPoint1 := ht.OpenChannel(
 		alice, bob, lntest.OpenChannelParams{
-			Amt: firstChanSize,
+			Amt:  firstChanSize,
+			Memo: "bob is a good peer",
```
Comment (yyforyongyu) [inline itest/lnd_onchain_test.go:470]: nit: make the memo a const?

**Review by yyforyongyu — APPROVED** [review]
Thanks for the PR!

File: itest/lnd_onchain_test.go (hunk 1 above)
Comment (guggero) [reply itest/lnd_onchain_test.go:470]: Yes a constant `testMemo` would shorten quite a few things in the itest.

**Review by guggero — APPROVED** [review]
LGTM, thanks a lot!

cc @saubyk

File: itest/lnd_onchain_test.go (hunk 1 above)
Comment (shaurya947) [reply itest/lnd_onchain_test.go]: Good idea, updated :slightly_smiling_face:

**Comment by saubyk** [comment]
> LGTM, thanks a lot!
> 
> cc @saubyk

Concept Ack

**Review by guggero — APPROVED** [review]

**Comment by shaurya947** [comment]
Apologies for the late turnaround here, I didn't realize a merge conflict had appeared. Just fixed, should be all good now :crossed_fingers:

**Review by positiveblue — APPROVED** [review]
Thank you for this @shaurya947 I think we are good to go :rocket:

**Pull request merged**